control.lua

```diff
@@ -9,11 +9,25 @@ bucket_settings = train_buckets(settings.startup["graftorio2-train-histogram-buc
 nth_tick = settings.startup["graftorio2-nth-tick"].value
 server_save = settings.startup["graftorio2-server-save"].value
 disable_train_stats = settings.startup["graftorio2-disable-train-stats"].value
+disable_per_player_stats = settings.startup["graftorio2-disable-per-player-stats"].value
 
 gauge_tick = prometheus.gauge("factorio_tick", "game tick")
 gauge_connected_player_count = prometheus.gauge("factorio_connected_player_count", "connected players")
 gauge_total_player_count = prometheus.gauge("factorio_total_player_count", "total registered players")
 
+gauge_player_connected =
+	prometheus.gauge("factorio_player_connected", "connected players by name", { "name" })
+gauge_player_last_online =
+	prometheus.gauge("factorio_player_last_online", "last tick the player was online", { "name" })
+gauge_player_time_online =
+	prometheus.gauge("factorio_player_time_online", "total seconds player spent online", { "name" })
```

Comment:
i believe it should be a counter and not a gauge, the value can only go up

Reply:
sure, I was just matching the other calls - as far as I could tell (although I did not look at every file), all other currently added metrics are either gauges or histograms (even `gauge_tick`, of which these two are kind of 'subsets', so to speak).

Reply:
Also on that note, `gauge_player_time_online` and `gauge_tick` should be actively reset to 0 on start if they are turned into counters, no? Otherwise if the server crashes without saving, but still exported its metrics at least once since the last save, the observed value could appear to decrease on the real-world timescale, even if it is monotonically increasing in relation to game-ticks.

Another interpretation (aside from just "this case is irrelevant") could be that `rate(factorio_tick)` should not be used, because the value can, exactly as stated by its gauge nature, go down?

I will wait for further confirmation on this before changing the metrics to counters. If you want I can also go over the existing metrics and change their type where appropriate, although I dont know how well prometheus handles existing metrics changing their type.